core/pkg/service/sync/sync_metrics.go

```diff
@@ -0,0 +1,35 @@
+package sync
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
```

Comment:
can we use OTel for exposing metrics and configure prometheus as the default exporter? this way we can change the backend implementation more easily if necessary.